src/components/Icon/CustomSvgIcon/CustomSvgIcon.tsx

```diff
@@ -1,25 +1,39 @@
-/* eslint-disable react/jsx-props-no-spreading */
-import React, { useCallback } from "react";
+import React, { FunctionComponent, useCallback, AriaRole, Ref } from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
 import SVG from "react-inlinesvg";
-import useIconScreenReaderAccessProps from "../../hooks/useIconScreenReaderAccessProps";
+import useIconScreenReaderAccessProps from "../../../hooks/useIconScreenReaderAccessProps";
+import VibeComponentProps from "../../../types/VibeComponentProps";
+import VibeComponent from "../../../types/VibeComponent";
 
-function modifySvgCode(svg, color = "currentColor") {
+function modifySvgCode(svg: string, color: string = "currentColor") {
   return svg.replace(/fill=".*?"/g, `fill="${color}"`);
 }
 
-const CustomSvgIcon = ({
-  className,
-  src,
+interface CustomSvgIconProps extends VibeComponentProps {
+  onClick?: (event: React.MouseEvent) => void;
+  src: string | object;
+  ariaLabel?: string;
+  tabIndex?: number;
+  role?: AriaRole;
+  ariaHidden?: boolean;
+  clickable?: boolean;
+  replaceToCurrentColor?: boolean;
+  customColor?: string;
+  ref?: Ref<SVGElement>;
+}
+
+const CustomSvgIcon: FunctionComponent<CustomSvgIconProps> = ({
+  className = "",
```

Comment:
No need for default value, `classnames` ignores `undefined`.

Reply:
I know, added for readability but I will remove